torch/_dynamo/optimizations/torchxla_integration.py

```diff
@@ -73,12 +72,11 @@ def import_torchxla():
     import torch_xla.debug.metrics as metrics
 
 
-def extract_compiled_graph(model: torch.fx.GraphModule, example_inputs):
+def extract_compiled_graph(xla_model: torch.fx.GraphModule, xla_args):
```

Comment:
is this a good place to do an assert that all xla_args and xla_model are on xla device?  Did you do that somewhere else (i may have missed it if it's not highlighted as a diff here)

Reply:
I believe there is no redundant checks in codepath reach here. We can add that.